test_util/mock_time_env.h

```diff
@@ -41,6 +41,30 @@ class MockTimeEnv : public EnvWrapper {
     current_time_ = time;
   }
 
+  // TODO: this is a workaround for the different behavior on different platform
+  // for timedwait timeout. Ideally timedwait API should be moved to env.
+  // details: PR #7101.
+  void InstallTimeWaitCallbackForMACOS() {
+#if defined(OS_MACOSX) && !defined(NDEBUG)
```

Comment:
Nice compromise. :) Two concerns:

That this function only needs to fix something on MacOS is an implementation detail. (Perhaps `InstallTimeWaitFixCallback`?)

Related to that, we could easily see different test behavior by platform due to this function having no effect on SyncPoints on most platforms and resetting them on MacOS. As much as possible, behavior should be consistent across platforms, with everything except SetCallBack outside of the ifdef (as I did in SafeMockTimeEnv).

Reply:
Make sense. Updated.